### ISSUE TITLE
SortO - sort columns by a different value from oAccessor

### ISSUE THREAD

**Issue by strictlymomo** [emeeks/semiotic#287]
I am implementing a diverging stacked bar chart using the recommended verbose data structure and would like to sort the columns by a properties in the ordinal value's parent object.

```
// e.g.,
myData=[ 
    {"Name": "Caesar", "Type": "Yay", "Value": 10, "Age": 40}, 
    {"Name": "Caesar", "Type": "Nay", "Value": 0, "Age": 40}, 
    {"Name": "Caligula", "Type": "Yay", "Value": 10, "Age": 20}, 
    {"Name": "Caligula", "Type": "Nay", "Value": -5, "Age": 20}.
    {"Name": "Alexander", "Type": "Yay", "Value": 10, "Age": 20}, 
    {"Name": "Alexander", "Type": "Nay", "Value": -1, "Age": 20}
]

<OrdinalFrame 
   data={myData}
   oAccessor={d => d["Name"]}
   rAccessor={"value"}
   sortO={(a,b) => a["Age"] < b["Age"]}  // wish a,b could reference d as opposed to d["Name"]
/>
```
While the [docs](https://emeeks.github.io/semiotic/#/semiotic/creatingbarchart) state 
> One thing to notice is that the sorting of items in OrdinalFrame is based on the sorting of the data array you send in, so if you want certain columns or pieces to appear first, then pre-sort your data,

the presorting is not working well due to properties with the same value in the data set.  Is it possible to access the parent object referenced via the ordinal value?  Or how about a column-sorting feature like `sortR`?

**Comment by emeeks** [comment]
Yes you're right, this needs to be extended to reveal column (and for sortR piece) data attributes. I'll get on this.

**Comment by strictlymomo** [comment]
Thank you for the prompt reply.  I'm really enjoying working with Semiotic and looking forward to this feature!

**Comment by emeeks** [comment]
Okay so in the next version sortO will send the same first two values and then the arrays of data for each column.

Just keep in mind that each column/row is an array of your data, so in your example since there's a 1-to-1 correspondence between name and age you can just use the first value in the array.

`sortO={(a, b, c, d) => d[1].age - c[1].age}`

I'll update this and close it when this is released.

**Comment by strictlymomo** [comment]
Wonderful, thank you so much 🙏.

On a related subject, the ordinal heatmap example on your interactive demo
site — 7 days, i %12 weeks — loops through the data “horizontally” across
weeks, rather than starting on Monday Week 1 (top left corner of grid),
descending towards Sunday (bottom left corner), and then onto Monday of
Week 2... it’s like the grid needs to rotate3D across the z-axis for r: ( )
=> 1 to iterate in a way suggested by the labels.

If you successively hover over the <g> DOM elements in dev tools, you’ll
see what I’m referring to.

Anyways, thanks again for all of your contributions. I appreciate your work
and perspective!

**Comment by emeeks** [comment]
Oh thanks, that was put together a long time ago and I should revisit it.

The sorting functionality is now available in v1.11.14 let me know if you run into any issues.
